lib/segment/src/index/struct_payload_index.rs

```diff
@@ -512,6 +511,36 @@ impl PayloadIndex for StructPayloadIndex {
         Ok(())
     }
 
+    fn set_indexed(
+        &mut self,
+        field: PayloadKeyTypeRef,
+        payload_schema: impl Into<PayloadFieldSchema>,
+        hw_counter: &HardwareCounterCell,
+    ) -> OperationResult<()> {
+        let payload_schema = payload_schema.into();
+
+        self.drop_index_if_incompatible(field, &payload_schema)?;
```

Comment:
Did you manually test that the index is properly created on restart if we crash just after deleting the incompatible index?

Reply:
I did